prow/plank/controller.go

```diff
@@ -142,6 +153,24 @@ func (c *Controller) Sync() error {
 	return fmt.Errorf("errors syncing: %v, errors reporting: %v", syncErrs, reportErrs)
 }
 
+func (c *Controller) syncProwJob(wg *sync.WaitGroup, jobs <-chan kube.ProwJob, syncErrors chan<- error, pm map[string]kube.Pod, pending map[string][]kube.ProwJob) {
+	wg.Add(1)
+	defer wg.Done()
+	for pj := range jobs {
+		if pj.Spec.Agent == kube.KubernetesAgent {
+			if err := c.syncKubernetesJob(pj, pm, pending); err != nil {
+				syncErrors <- err
+			}
+		} else if pj.Spec.Agent == kube.JenkinsAgent {
+			if err := c.syncJenkinsJob(pj, pending); err != nil {
+				syncErrors <- err
+			}
+		} else {
+			syncErrors <- fmt.Errorf("job %s has unsupported agent %s", pj.Metadata.Name, pj.Spec.Agent)
```

Comment:
I mean it should not be regarded as an error here. It's definitely not unrelated if we change the controller to use informers and resync prowJobs on errors.